File: manaxc-project/code/importers/get_athlete_best_times_schema.py
#!/usr/bin/env python3
"""Get the schema for athlete_best_times table."""

import os
from dotenv import load_dotenv
from supabase import create_client

# Load environment
load_dotenv('../../website/.env.local')

supabase = create_client(
    os.getenv('NEXT_PUBLIC_SUPABASE_URL'),
    os.getenv('NEXT_PUBLIC_SUPABASE_ANON_KEY')
)

# Get one row to see all columns
result = supabase.table('athlete_best_times').select('*').limit(1).execute()

if result.data:
    print("Columns in athlete_best_times table:")
    for column in sorted(result.data[0].keys()):
        print(f"  - {column}")
else:
    print("No rows in athlete_best_times table")

File: manaxc-project/website/get_athlete_best_times_schema.js
import { createClient } from '@supabase/supabase-js'
import dotenv from 'dotenv'

dotenv.config({ path: '.env.local' })

const supabase = createClient(
  process.env.NEXT_PUBLIC_SUPABASE_URL,
  process.env.SUPABASE_SERVICE_ROLE_KEY
)

async function main() {
  // Get a sample row to see the schema
  const { data, error } = await supabase
    .from('athlete_best_times')
    .select('*')
    .limit(1)

  if (error) {
    console.error('Error:', error)
    return
  }

  console.log('athlete_best_times columns:')
  if (data && data.length > 0) {
    console.log(Object.keys(data[0]))
  } else {
    console.log('No data in table')
  }
}

main()
